dev-tools/prepare_release_candidate.py

```diff
@@ -0,0 +1,271 @@
+      raise RuntimeError('pending coming[%s] documentation changes found in %s' % (release_version, pending_files))
+
+
+  run('cd dev-tools && mvn versions:set -DnewVersion=%s -DgenerateBackupPoms=false' % (release_version))
+  run('cd rest-api-spec && mvn versions:set -DnewVersion=%s -DgenerateBackupPoms=false' % (release_version))
+  run('mvn versions:set -DnewVersion=%s -DgenerateBackupPoms=false' % (release_version))
+
+  remove_version_snapshot(VERSION_FILE, release_version)
+
+  print('*** Done removing snapshot version. DO NOT COMMIT THIS, WHEN CREATING A RELEASE CANDIDATE.')
+
+  shortHash = subprocess.check_output('git log --pretty=format:"%h" -n 1', shell=True).decode('utf-8')
+  localRepo = '/tmp/elasticsearch-%s-%s' % (release_version, shortHash)
+  localRepoElasticsearch = localRepo + '/org/elasticsearch'
+  if os.path.exists(localRepoElasticsearch):
+    print('clean local repository %s' % localRepoElasticsearch)
+    shutil.rmtree(localRepoElasticsearch)
+
+  if install_only:
+    mvn_targets = 'install'
+  else:
+    mvn_targets = 'install deploy'
+  install_command = 'mvn clean %s -Prelease -Dskip.integ.tests=true -Dgpg.keyname="%s" -Dpackaging.rpm.rpmbuild=/usr/bin/rpmbuild -Drpm.sign=true -Dmaven.repo.local=%s -Dno.commit.pattern="\\bno(n|)commit\\b" -Dforbidden.test.signatures=""' % (mvn_targets, gpg_key, localRepo)
+  clean_repo_command = 'find %s -name _remote.repositories -exec rm {} \;' % (localRepoElasticsearch)
+  rename_metadata_files_command = 'for i in $(find %s -name "maven-metadata-local.xml*") ; do mv "$i" "${i/-local/}" ; done' % (localRepoElasticsearch)
+  s3_sync_command = 's3cmd sync %s s3://download.elasticsearch.org/elasticsearch/staging/elasticsearch-%s-%s/org/' % (localRepoElasticsearch, release_version, shortHash)
+  s3_bucket_sync_to = 'download.elasticsearch.org/elasticsearch/staging/elasticsearch-%s-%s/repos' % (release_version, shortHash)
+  build_repo_command = 'dev-tools/build_repositories.sh %s' % (release_version)
+  if install_and_deploy:
+    for cmd in [install_command, clean_repo_command]:
+      run(cmd)
+    rename_local_meta_files(localRepoElasticsearch)
+  else:
+    print('')
+    print('*** To create a release candidate run: ')
+    print('  %s' % (install_command))
+    print('  1. Remove all _remote.repositories: %s' % (clean_repo_command))
+    print('  2. Rename all maven metadata files: %s' % (rename_metadata_files_command))
+  if push:
+    run(s3_sync_command)
+    env_vars = {'S3_BUCKET_SYNC_TO': s3_bucket_sync_to, 'S3_BUCKET_SYNC_FROM' : s3_bucket_sync_to}
+    run(build_repo_command, env_vars)
+  else:
+    print('')
+    print('*** To push a release candidate to s3 run: ')
+    print('  1. Sync %s into S3 bucket' % (localRepoElasticsearch))
+    print ('    %s' % (s3_sync_command))
+    print('  2. Create repositories: ')
+    print ('    export S3_BUCKET_SYNC_TO="%s"' % (s3_bucket_sync_to))
+    print ('    export S3_BUCKET_SYNC_FROM="%s"' % (s3_bucket_sync_to))
```

Comment:
I thought about this and it is wrong. We need to specify the existing official repo here. So, if we release 2.0.0 we retain the beta release in there. Just remove this line and the `download.elasticsearch.co` is used to sync and all works as expected.